### PR TITLE
Update template

### PR DESCRIPTION
**Pull request by elijahchancey**
I copied and pasted from Buildkite's regular elastic template and made this template a bit more "standardized"

Update default Host family & Instance Type to mac2
Rename BuildkiteAgentQueue to BuildkiteQueue (reduce delta between templates)
Add Paramater for EC2 Key Name
Add Cost Allocation Tags
Add EC2/ASG Tags
Add InstanceName

### CODE REVIEW

File: template.yml [hunk 1]
```diff
@@ -49,14 +49,50 @@ Parameters:
     NoEcho: true
     Description: "Buildkite Agent token from https://buildkite.com/organizations/-/agents"
 
-  BuildkiteAgentQueue:
+  BuildkiteQueue:
```
Comment (matthewborden) [inline template.yml]: This is a great suggestion for consistency between our elastic stack templates. However, it's not a backwards compatible change for existing consumers of the template.

**Review by matthewborden — DISMISSED** [review]
Thanks for the contribution! We'd love to merge this once the rename of the queue name parameter has been reverted.

File: template.yml (hunk 1 above)
Comment (elijahchancey) [reply template.yml]: fixed. @matthewborden

**Comment by elijahchancey** [comment]
reverted.

**Review by moskyb — APPROVED** [review]
thank you @elijahchancey! love your work ^_^

**Comment by moskyb** [comment]
i did not know that that was going to be public 💀

i still love your work @elijahchancey, i still value you @matthewborden 💚

**Comment by moskyb** [comment]
who knew you can't edit review dismissals?

**Pull request merged**